nfs/src/rpc_task.cpp

```diff
@@ -1422,11 +1422,18 @@ void rpc_task::run_create_file()
         CREATE3args args;
         ::memset(&args, 0, sizeof(args));
 
+        const fuse_ctx *ctx = fuse_req_ctx(get_fuse_req());
+        assert(ctx != nullptr);
+
         args.where.dir = get_client()->get_nfs_inode_from_ino(parent_ino)->get_fh();
         args.where.name = (char*)rpc_api->create_task.get_file_name();
         args.how.mode = (rpc_api->create_task.get_fuse_file()->flags & O_EXCL) ? GUARDED : UNCHECKED;
         args.how.createhow3_u.obj_attributes.mode.set_it = 1;
         args.how.createhow3_u.obj_attributes.mode.set_mode3_u.mode = rpc_api->create_task.get_mode();
```

Comment:
I think we must also subtract umask from the mode.
Can you experiment with setting umask to a different value and see if we correctly honor it